src/Modal.test.tsx

```diff
@@ -38,19 +45,14 @@ test("the modal can be closed by clicking the close button", () => {
       </Modal>
     );
   }
-  const { queryByRole, getByLabelText } = render(<ModalFn />);
-  act(() => {
-    getByLabelText("action.close").click();
+  const user = userEvent.setup();
+  const { queryByRole, getByRole } = render(<ModalFn />);
+  await act(async () => {
+    await user.click(getByRole("button", { name: "action.close" }));
```

Comment:
There shouldn't need to be an `act` here as user-event handles calling `act` internally (though it has been known to be brittle at doing that in the past…)